clients/rest/rest.go

```diff
@@ -214,3 +257,21 @@ func (c *Client) Close() (err error) {
 	c.httpClient.CloseIdleConnections()
 	return
 }
+
+func (c *Client) setTlSConfig() (*tls.Config, error) {
```

Comment:
This looks like a getMethod but also sets its.

May be we can simplify with

1 Just keep the TLS config separate 
2 Call a prepareTransport func in execute to just set at once